skorch/hf.py

```diff
@@ -1004,3 +1006,182 @@ def _step_optimizer(self, step_fn):
     def on_train_end(self, net, X=None, y=None, **kwargs):
         super().on_train_end(net, X=X, y=y, **kwargs)
         self.module_ = self.accelerator.unwrap_model(self.module_)
+
+
+class HfHubStorage:
+    """Helper class that allows writing data to the Hugging Face Hub.
+
+    Use this, for instance, in combination with checkpoint callbacks such as
+    :class:`skorch.callbacks.training.TrainEndCheckpoint` or
+    :class:`skorch.callbacks.training.Checkpoint` to upload the trained model
+    directly to the Hugging Face Hub instead of storing it locally.
+
+    To use this, it is necessary to install the `Hugging Face Hub library
+    <https://huggingface.co/docs/huggingface_hub/index>`__.
+
+    .. code:: bash
+
+        python -m pip install huggingface_hub
+
+    Note that writes to the Hub are synchronous. Therefore, if the time it takes
+    to upload the data is long compared to training the model, there can be a
+    signficant slowdown. It is best to use this with
+    :class:`skorch.callbacks.training.TrainEndCheckpoint`, as that checkpoint
+    only uploads the data once, at the end of training. Also, using this writer
+    with :class:`skorch.callbacks.training.LoadInitState` is not supported for
+    now.
+
+    Parameters
+    ----------
+    hf_api : instance of huggingface_hub.HfApi
+      Pass an instantiated ``huggingface_hub.HfApi`` object here.
+
+    path_in_repo : str
+      The name that the file should have in the repo, e.g. ``my-model.pkl``. If
+      you want each upload to have a different file name, instead of overwriting
+      the file, use a templated name, e.g. ``my-model-{}.pkl``. Then your files
+      will be called ``my-model-1.pkl``, ``my-model-2.pkl``, etc.
```

Comment:
This should note that this does not regard the previous contents of the repository. I think this should be made clear to the user